README.md

```diff
@@ -282,6 +285,8 @@ Some Serilog packages require a reference to a logger configuration object. The
 
 When the configuration specifies a discrete value for a parameter (such as a string literal), the package will attempt to convert that value to the target method's declared CLR type of the parameter. Additional explicit handling is provided for parsing strings to `Uri`, `TimeSpan`, `enum`, arrays and custom collections.
 
+Since version 3.5.0, conversion will use the invariant culture (`CultureInfo.InvariantCulture`) as long as the `ReadFrom.Configuration(IConfiguration configuration, ConfigurationContext context)` method is used. Obsolete methods use the current culture to preserve backward compatibility.
```

Comment:
A lot of people will still want to write `ReadFrom.Configuration(config)` and skip the redundant extra argument. A breaking change might be fair, here, given that the source for most settings coming from configuration will be JSON, where anything other than the invariant culture is probably just harboring bugs.

What do you think?

Reply:
Actually, the newly introduced method already supports calling `ReadFrom.Configuration(config)` without an extra argument since the `context` parameter defaults to `null`:

```csharp
public static LoggerConfiguration Configuration(
    this LoggerSettingsConfiguration settingConfiguration,
    IConfiguration configuration,
    ConfigurationContext context = null)
```

By the way, I had to remove the default `= null` in the overload taking a `DependencyContext`. That's what I tried to explain in my original message but was maybe not very clear.

```csharp
public static LoggerConfiguration Configuration(
    this LoggerSettingsConfiguration settingConfiguration,
    IConfiguration configuration,
    DependencyContext dependencyContext)
```

So people just using `ReadFrom.Configuration(config)` will opt in the new behaviour (invariant culture) by updating to `Serilog.Settings.Configuration` 3.5.0 and recompiling their app. The only downside is that they will probably not even realise they opt in the new behaviour.

And people using `ReadFrom.Configuration(config, … extra argument …)` will get an obsolete warning and will be forced to have a look at the new overload and think about how they want to create the new `ConfigurationContext` object.

Reply:
Ah, thanks - I missed that. I'm trying to get a few minutes spare again to catch up with this, I'll try to loop back later today :-)